posthog/models/team.py

```diff
@@ -65,14 +65,22 @@ def create(self, *args, **kwargs) -> "Team":
             raise ValueError("Creating organization-less projects is prohibited")
         return super().create(*args, **kwargs)
 
-    def get_team_from_token(self, token: Optional[str]) -> Optional["Team"]:
+    def get_team_from_token(self, token: Optional[str], only: Tuple) -> Optional["Team"]:
         if not token:
             return None
         try:
-            return Team.objects.get(api_token=token)
-        except Team.DoesNotExist:
+            return Team.objects.filter(api_token=token).only(*only)[0]
```

Comment:
Nit: No try-catch needed, this could be `Team.objects.filter(api_token=token).only(*only).first()` which would give the same result.

Reply:
Agreed here. Try/Catches are slow in Python so any way we can remove it would be nice.